### PR TITLE
test(addfront):add test & solution

### PR DESCRIPTION
**Pull request by Hamzaelkhatri**

### CODE REVIEW

File: tests/addfront_test/main.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+func main() {
+	table := [][]string{
+		{"", "a", "b", "c"},
+		{"a", "b", "c", "d"},
+		{"Hello", "World", "!"},
+		{"Hello", "World", "!"},
+		{"Hello", ""},
+		{"0", "1", "2", "3", "4", "5", "6", "7", "8", "9"},
+	}
+	for _, arg := range table {
```
Comment (eslopfer) [inline tests/addfront_test/main.go:19]: When I run this test I get the following error:
```console
tests/addfront_test/main.go:20:83: too many arguments in call to challenge.Function
        have (string, func(str string, slice []string) []string, func(str string, slice []string) []string, string, []string)
        want (string, interface{}, interface{}, ...interface{})
  ```

File: tests/addfront_test/main.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+)
+
+func main() {
+	table := [][]string{
+		{"", "a", "b", "c"},
+		{"a", "b", "c", "d"},
+		{"Hello", "World", "!"},
+		{"Hello", "World", "!"},
+		{"Hello", ""},
+		{"0", "1", "2", "3", "4", "5", "6", "7", "8", "9"},
```
Comment (MSilva95) [inline tests/addfront_test/main.go:17]: add random test
```
		{random.Str(chars.Words, 6), "Hello", ""},

File: solutions/addfront.go [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package solutions
```
Comment (MSilva95) [inline solutions/addfront.go:1]: Don't forget to fix your solution to implement the changes requested in the readme.

**Review by MSilva95 — CHANGES_REQUESTED** [review]

File: solutions/addfront.go (hunk 3 above)
Comment (Hamzaelkhatri) [reply solutions/addfront.go:1]: i added

**Review by MSilva95 — CHANGES_REQUESTED** [review]

File: solutions/addfront.go [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+
+func AddFront(str string, slice []string) []string {
```
[the remark this answers is not in the document]
Comment (Hamzaelkhatri) [reply solutions/addfront.go:3]: the function `fmt.Println` print space after each element of the array expect the last one, so in this case `[]string{"", "a", "b", "c"}`  **Println** print space  after the (empty) element . 
if we use, for example : 
```go 
result := AddFront("Hello", []string{"", "a", "b", "c"})
for _, str := range result {
       fmt.Println("[" + str + "]")
}
```
and the output 
```console 
$ go run  . 
[Hello]
[]
[a]
[b]
[c]
```
The solution's output adds two spaces between Hello and a.

**Review by MSilva95 — APPROVED** [review]

**Pull request merged**